pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -286,7 +287,22 @@ func (f *ring1Factory) Scaler(mapping *meta.RESTMapping) (kubectl.Scaler, error)
 	if err != nil {
 		return nil, err
 	}
-	return kubectl.ScalerFor(mapping.GroupVersionKind.GroupKind(), clientset)
+
+	// create scales getter
+	discoClient, err := f.clientAccessFactory.DiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	restClient, err := f.clientAccessFactory.RESTClient()
+	if err != nil {
+		return nil, err
+	}
+	mapper, _ := f.Object()
+	resolver := scaleclient.NewDiscoveryScaleKindResolver(discoClient)
+	scalesGetter := scaleclient.New(restClient, mapper, dynamic.LegacyAPIPathResolverFunc, resolver)
+	gvk := mapping.GroupVersionKind.GroupVersion().WithResource(mapping.Resource)
```

Comment:
I suspect we want a factory method that gives back the `scalesGetter`

Reply:
alright, thanks.